src/common/perf_counters.cc

```diff
@@ -194,6 +194,11 @@ void PerfCounters::inc(int idx, uint64_t amt)
   if (data.type & PERFCOUNTER_LONGRUNAVG) {
     data.avgcount++;
     data.u64 += amt;
+    uint64_t m, m_new;
+    do {
+      m = data.max_u64.load();
+      m_new = std::max(amt, m);
```

Comment:
we may want to save the atomic operation if there was no change in the max (i.e. if m_new == m)?